bin/src/command/orders.rs

```diff
@@ -1262,35 +1206,24 @@ impl CommandServer {
         for ref mut worker in self.workers.iter_mut().filter(|worker| {
             worker.run_state != RunState::Stopping && worker.run_state != RunState::Stopped
         }) {
-            // sort out the specifically targeted worker, if provided
-            if let Some(id) = worker_id {
-                if id != worker.id {
-                    continue;
-                }
-            }
-
-            let should_stop_worker =
-                order == ProxyRequestOrder::SoftStop || order == ProxyRequestOrder::HardStop;
+            let should_stop_worker = order == Order::SoftStop || order == Order::HardStop;
             if should_stop_worker {
                 worker.run_state = RunState::Stopping;
                 stopping_workers.insert(worker.id);
             }
 
-            // let request_id = request_identifier.to_worker_request_id();
-            let req_id = format!("{}-worker-{}", request_identifier.client, worker.id);
+            let req_id = format!("{}-worker-{}", client_id, worker.id);
             worker.send(req_id.clone(), order.clone()).await;
             self.in_flight.insert(req_id, (worker_order_tx.clone(), 1));
 
             found = true;
             worker_count += 1;
         }
 
-        let should_stop_main = (order == ProxyRequestOrder::SoftStop
```

Comment:
Are you sure about this change?
Same could you use `matches!`?

Reply:
Yes I'm sure about the change. Removing the optional `worker_id` from the Request is a key feature of the PR. The only Request that can be directed to a specific worker is of type `UpgradeWorker(worker_id)`.

I'd like to implement a custom method on Request, something like `.is_a_stop()`